---
title: "gtf"
output: html_document
date: "2023-05-25"
author: "AÃ¯sha Schuhegger"
note: "run the code discretized_form.Rmd before running this code" 
---

#gtf without NA rows
#gtf only SE AS
#gtf only RI AS 


Load libraries
```{r}
library(data.table)
```

Create data frame that contains the necessary information (events belonging to the Alternatively Spliced Cluster of RI and SE) regarding the gtf annotation format.  
```{r}
events <- fread("/nfs/home/students/a.schuhe/template/events.csv.gz")

events_AS <- events[events$event_id %in% rownames_discretized_df, ]
events_AS <- events_AS[order(events_AS$event_id), ]

helper_eventID <- data.frame(matrix(nrow=nrow(events_AS), ncol=2))
colnames(helper_eventID) <- c("event_id", "ID")
helper_eventID$event_id <- events_AS$event_id
helper_eventID$ID <- rownames(helper_eventID)

gtf <- data.frame(matrix(nrow=nrow(events_AS), ncol=9))
# Set column names
colnames(gtf) <- c("seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attribute")

#"empty" columns should be denoted with a '.'
#seqname - name of the chromosome or scaffold; chromosome names can be given with or without the 'chr' prefix. 
#source - name of the program that generated this feature, or the data source (database or project name)
#feature - feature type name, e.g. Gene, Variation, Similarity
#start - Start position* of the feature, with sequence numbering starting at 1.
#end - End position* of the feature, with sequence numbering starting at 1.
#score - A floating point value.
#strand - defined as + (forward) or - (reverse).
#frame - One of '0', '1' or '2'. '0' indicates that the first base of the feature is the first base of a codon, '1' that the second base is the first base of a codon, and so on..
#attribute - A semicolon-separated list of tag-value pairs, providing additional information about each feature.

gtf$source <- "IHEC"
helper_eventID$ID <- as.integer(helper_eventID$ID)
for (i in 1:nrow(gtf)){
  event_id <- events_AS$event_id[i]
  #id_value <- helper_eventID$ID[helper_eventID$event_id == event_id]
  id_value <- sprintf("%015d", helper_eventID$ID[helper_eventID$event_id == rownames(temp_df[i,])])
  values <- strsplit(event_id, ";")[[1]]
  separated_values <- unlist(strsplit(values, ":", fixed=TRUE))
  separated_values[4] <- strsplit(as.character(separated_values[4]), "-", fixed = TRUE)
  #separated_values[[4]][1] #SE: e1, RI: s1
  #separated_values[[4]][2] #SE: s2, RI: e1
  separated_values[5] <- strsplit(as.character(separated_values[5]), "-", fixed = TRUE)
  #separated_values[[5]][1] #SE: e2, RI: s2
  #separated_values[[5]][2] #SE: s3, RI: e2
  
  gtf$seqname[i] <- separated_values[3]
  gtf$feature[i] <- 'gene'#actually 'exon' but STITCHIT has to 'think' that it is a gene 
  if(separated_values[2] == "SE") {
    gtf$start[i] <- separated_values[[4]][2] #SE: s2 
    gtf$end[i] <- separated_values[[5]][1] #SE: e2
  } else { #RI
    gtf$start[i] <- separated_values[[4]][2] # RI: e1
    gtf$end[i] <- separated_values[[5]][1] #RI: s2
  }
  gtf$score[i] <- '.'
  gtf$strand[i] <- separated_values[6]
  gtf$frame[i] <- '.'
  gtf$attribute[i] <- sprintf('gene_id "%s"; event_type "%s";', id_value, separated_values[2]) #gene_id = event_id and ";" is transformed to "_" and Skipped Exon or Retained Intron 

  
}

```

Writing the gtf file with the help of the data frame 'gtf'
```{r}
#gtf <- as.character(gtf)
gtf <- apply(gtf,2,as.character)
write.table(gtf, file = "annotation_events.gtf", sep = "\t", quote = FALSE, row.names = FALSE)

```

